components/pd_client/src/util.rs

```diff
@@ -170,21 +174,33 @@ impl LeaderClient {
     ///
     /// If `force` is false, it will reconnect only when members change.
     pub async fn reconnect(&self, force: bool) -> Result<()> {
-        let (future, start) = {
+        let start = Instant::now();
+
+        let future = {
             let inner = self.inner.rl();
-            if inner.last_update.elapsed() < Duration::from_secs(RECONNECT_INTERVAL_SEC) {
+            if inner.last_update.elapsed() < RECONNECT_UPDATE_INTERVAL {
                 // Avoid unnecessary updating.
                 return Ok(());
             }
-
-            let start = Instant::now();
+            if !force && inner.last_try_reconnect.elapsed() < ALL_NON_FORCE_RECONNECT_INTERVAL {
```

Comment:
Why check `force`? It should probably always retry if timeout.

Reply:
In order to make this restriction only take effect for `reconnect` with `force=false`, I adjusted the level of `reconnect` initiated by region-related requests such as heartbeat requests.

Reply:
Two `reconnect`s with `force=false` that are executed need to be separated by more than 100ms, but `force=true` is not limited.

Reply:
I don't understand why `force` matters. What could go wrong if ignoring `false`? Will the problem be also solved?

Reply:
In order to control the scope of influence of this PR, it currently does not limit the retry of sync_request.

Reply:
They should also handle connection errors, and probably did.